src/sonic-yang-models/yang-models/sonic-syslog.yang

```diff
@@ -9,22 +9,64 @@ module sonic-syslog {
         prefix inet;
     }
 
+    import sonic-vrf {
+        prefix vrf;
+    }
+
+    description "Syslog YANG Module for SONiC OS: remote syslog logging";
+
+    revision 2022-04-18 {
+        description "Initial revision.";
+    }
+
+    typedef vrf-device {
+        description "Represents syslog VRF device";
+        type enumeration {
+            enum default;
+            enum mgmt;
+        }
+    }
+
     container sonic-syslog {
 
         container SYSLOG_SERVER {
 
-            description "SYSLOG SERVER part of config_db.json";
+            description "SYSLOG_SERVER part of config_db.json";
 
             list SYSLOG_SERVER_LIST {
 
                 key "server_address";
 
                 leaf server_address {
-                    description "syslog server ip address";
+                    description "Syslog server IP address";
                     type inet:ip-address;
                 }
-            } /* end of list SYSLOG_SERVER_LIST */ 
 
-        } /* end of container SYSLOG_SERVER */
+                leaf source {
+                    description "Syslog source IP address";
```

Comment:
What happens when the source IP is not configured within a VRF (say. mgmt or data VRF)? rsyslog takes available IP from the kernel and send the log messages?

Reply:
SONIC YANG model will be used for validating the configs present in json file for config load/reload, so YANG validation is important.

Reply:
@venkatmahalingam handled

Reply:
@venkatmahalingam such kind of validation is not possible with the current DB schema and YANG approach since we don't reference a configured IPs. Also not sure if it's even possible to add such a constraint in current YANG model to validate whether interface IP is configured across the specific VRF. This brings a lot of complexity. Any suggestions are appreciated.